system.py

```diff
@@ -166,10 +166,11 @@ def Detect(modeltype,goaltype,csvfile,dof,effdof,orderuse,reg=True,lmax=3,alpha=
 	print('Here python!')
 	print(type(modeltype),type(goaltype),type(csvfile),type(dof),type(effdof),type(orderuse))
 	data_ = pd.read_csv(csvfile, header=None).values
-	kstiff = data_[:,0].reshape((dof,1))
+	kstiff = data_[:,0]
+	kstiff = np.diagflat(kstiff)
 	mass = data_[:,1]
 	modedata = data_[:,2:2+len(effdof)]
-	modedata = modedata[[x-1 for x in orderuse],:].reshape((len(orderuse),))
+	modedata = modedata[[x-1 for x in orderuse],:].reshape((1,len(orderuse)))
```

Comment:
Why use the different shape? I found that the class using _modedata_ had no respective modifications.